src/libgap-api.c

```diff
@@ -42,46 +68,135 @@ Obj GAP_ValueGlobalVariable(const char * name)
     }
 }
 
-//
-// Evaluate a string of GAP commands
-//
-// To see an example of how to use this function
-// see tst/testlibgap/basic.c
-//
-Obj GAP_EvalString(const char * cmd)
+
+////
+//// arithmetic
+////
+
+int GAP_EQ(Obj a, Obj b)
 {
-    Obj instream;
-    Obj res;
-    Obj viewObjFunc, streamFunc;
+    return EQ(a, b);
+}
 
-    streamFunc = GAP_ValueGlobalVariable("InputTextString");
-    viewObjFunc = GAP_ValueGlobalVariable("ViewObj");
+int GAP_LT(Obj a, Obj b)
+{
+    return LT(a, b);
+}
 
-    instream = DoOperation1Args(streamFunc, MakeString(cmd));
-    res = READ_ALL_COMMANDS(instream, False, True, viewObjFunc);
-    return res;
+int GAP_IN(Obj a, Obj b)
+{
+    return IN(a, b);
+}
+
+Obj GAP_SUM(Obj a, Obj b)
+{
+    return SUM(a, b);
+}
+
+Obj GAP_DIFF(Obj a, Obj b)
+{
+    return DIFF(a, b);
+}
+
+Obj GAP_PROD(Obj a, Obj b)
+{
+    return PROD(a, b);
+}
+
+Obj GAP_QUO(Obj a, Obj b)
+{
+    return QUO(a, b);
+}
+
+Obj GAP_LQUO(Obj a, Obj b)
+{
+    return LQUO(a, b);
+}
+
+Obj GAP_POW(Obj a, Obj b)
+{
+    return POW(a, b);
+}
+
+Obj GAP_COMM(Obj a, Obj b)
+{
+    return COMM(a, b);
+}
+
+Obj GAP_MOD(Obj a, Obj b)
+{
+    return MOD(a, b);
+}
+
+
+////
+//// booleans
+////
+
+Obj GAP_True;
+Obj GAP_False;
+Obj GAP_Fail;
+
+
+////
+//// lists
+////
+
+int GAP_IsList(Obj obj)
+{
+    return obj && IS_LIST(obj);
+}
+
+Int GAP_LenList(Obj obj)
+{
+    return LEN_LIST(obj);
+}
+
+void GAP_AssList(Obj list, UInt pos, Obj val)
+{
+    if (val)
+        ASS_LIST(list, pos, val);
+    else
+        UNB_LIST(list, pos);
```

Comment:
Here, I though that unbinding (a rare operation, I believe) could be transparently achieved by the same function as is used for assignment. Good/bad idea?

Reply:
We discussed such behaviour before, where code like
```
list[5] := SomeFunction("foo");
```
was supposed to unbind `list[5]` if `SomeFunction` did not return a value (i.e. 0). I don't particularly like it.

Reply:
I don't think that's the same discussion at all. What you are talking about is a change on the GAP language level; and I agree that we rejected that idea, and I agree with that rejection.

There is no reason why this must dictate what the C API does, though. In practice, unbinding *is* the same as assigning NULL. Just because we don't expose this on the GAP language level does not justify that we must contort ourselves on the C level. That said, there may be valid reasons to reject this, but I don't see how what you mention is one. Perhaps if you elaborate as to what the actual problems you are concerned about are...?